src/provider/WalletLinkProvider.ts

```diff
@@ -269,6 +269,14 @@ export class WalletLinkProvider
     }
 
     const relay = await this.initializeRelay()
+
+    if (
+      !this._isAuthorized() &&
+      !relay.supportsUnauthedAddEthereumChain(chainId.toString())
```

Comment:
intent would be more clear to me if the code were written as

```
const isWhitelistedNetwork = this.ui.inlineAddEthereumChain(chainId)
if (!this._isAuthorized() && !isWhitelistedNetwork)
```

Reply:
Updated 👍